File: backend/src/main/java/com/leavemarker/entity/LeaveBalance.java
package com.leavemarker.entity;

import com.leavemarker.enums.LeaveType;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "leave_balances",
       uniqueConstraints = {
           @UniqueConstraint(columnNames = {"employee_id", "leave_type", "year"})
       })
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LeaveBalance extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    @Enumerated(EnumType.STRING)
    @Column(name = "leave_type", nullable = false, length = 30)
    private LeaveType leaveType;

    @Column(nullable = false)
    private Integer year;

    @Column(nullable = false)
    @Builder.Default
    private Double totalQuota = 0.0;

    @Column(nullable = false)
    @Builder.Default
    private Double used = 0.0;

    @Column(nullable = false)
    @Builder.Default
    private Double pending = 0.0;

    @Column(nullable = false)
    @Builder.Default
    private Double available = 0.0;

    @Column(nullable = false)
    @Builder.Default
    private Double carriedForward = 0.0;
}

File: backend-go/internal/models/leave_balance.go
package models

import "github.com/leavemarker/backend-go/internal/enums"

// LeaveBalance represents a leave balance entity
type LeaveBalance struct {
	BaseModel
	EmployeeID     uint            `gorm:"not null;index;uniqueIndex:idx_emp_type_year" json:"employeeId"`
	LeaveType      enums.LeaveType `gorm:"size:30;not null;uniqueIndex:idx_emp_type_year" json:"leaveType"`
	Year           int             `gorm:"not null;uniqueIndex:idx_emp_type_year" json:"year"`
	TotalQuota     float64         `gorm:"default:0.0" json:"totalQuota"`
	Used           float64         `gorm:"default:0.0" json:"used"`
	Pending        float64         `gorm:"default:0.0" json:"pending"`
	Available      float64         `gorm:"default:0.0" json:"available"`
	CarriedForward float64         `gorm:"default:0.0" json:"carriedForward"`

	// Relationships
	Employee Employee `gorm:"foreignKey:EmployeeID" json:"-"`
}

func (LeaveBalance) TableName() string {
	return "leave_balances"
}
